default-recommendations/for-loop-shortcuts-test.rkt

```diff
@@ -909,3 +909,73 @@ test: "(unless ...) in a for* loop refactored to #:when clause"
        #:unless (number? x))
   (displayln x))
 ------------------------------------------------------------
+
+
+test: "for/vector with in-range 0 n gets #:length n"
+------------------------------------------------------------
+(define n 5)
+(for/vector ([i (in-range 0 n)]) i)
+------------------------------------------------------------
+------------------------------------------------------------
+(define n 5)
+(for/vector #:length n
+            ([i (in-range 0 n)])
+  i)
+------------------------------------------------------------
+
+
+test: "for/vector with in-range n gets #:length n"
+------------------------------------------------------------
+(define n 5)
+(for/vector ([i (in-range n)]) i)
+------------------------------------------------------------
+------------------------------------------------------------
+(define n 5)
+(for/vector #:length n
+            ([i (in-range 0 n)])
+  i)
+------------------------------------------------------------
+
+
+test: "for*/vector with in-range 0 n gets #:length n"
+------------------------------------------------------------
+(define n 5)
+(for*/vector ([i (in-range 0 n)]) i)
+------------------------------------------------------------
+------------------------------------------------------------
+(define n 5)
+(for*/vector #:length n
+             ([i (in-range 0 n)])
+  i)
+------------------------------------------------------------
+
+
+test: "for/vector with literal end not refactorable"
```

Comment:
Here and in the other negative test cases you added below, instead of saying "not refactorable" in the test description, say "won't have #:length added". That makes it clearer that these test cases are specifically for the new rule you added, rather than any other existing refactoring rules related to `for/vector`.